neqo-udp/src/lib.rs

```diff
@@ -8,37 +8,133 @@
 #![allow(clippy::missing_panics_doc)] // Functions simply delegate to tokio and quinn-udp.
 
 use std::{
+    cell::RefCell,
     io::{self, IoSliceMut},
-    net::{SocketAddr, ToSocketAddrs},
+    net::SocketAddr,
     slice,
 };
 
-use neqo_common::{Datagram, IpTos};
+use neqo_common::{qdebug, Datagram, IpTos};
 use quinn_udp::{EcnCodepoint, RecvMeta, Transmit, UdpSocketState};
-use tokio::io::Interest;
 
 /// Socket receive buffer size.
 ///
 /// Allows reading multiple datagrams in a single [`Socket::recv`] call.
 const RECV_BUF_SIZE: usize = u16::MAX as usize;
```

Comment:
This is the max on all platforms?

Reply:
That seems unlikely.  But @larseggert, does that need to be fixed here, or can we track that with another issue?

Reply:
I think finding the right value needs more experimentation.

Here is what others use:

Google's Quiche handles [1472 bytes per segment](https://github.com/google/quiche/blob/36f574836fc8e8c9003ee5b08f037f998372b909/quiche/quic/core/quic_constants.h#L52) with at most [16 segments per read](https://github.com/google/quiche/blob/36f574836fc8e8c9003ee5b08f037f998372b909/quiche/quic/core/quic_packet_reader.h#L21).

MSQuic uses [`(UINT16_MAX - CXPLAT_UDP_HEADER_SIZE)`](https://github.com/microsoft/msquic/blob/8a741dc87331a346d38123b886331b240c06c509/src/platform/datapath_winuser.c#L51-L54).

Quinn uses [a receive buffer of `64 * 1024`](https://github.com/quinn-rs/quinn/blob/c445e9b8ff4026b1b27979f4c9c21127402df12a/quinn/src/endpoint.rs#L696-L701).

Note that in Firefox this buffer is allocated once per socket thread. Given that Firefox has a single socket thread only, there is effectively a single `u16::MAX` receive buffer only.

Would you recommend a larger buffer size to start with @larseggert?

Reply:
We can start with this, but maybe add a `TODO` to capture that we should experiment with what a good size would be.

Reply:
:+1: be05257e